### ISSUE TITLE
Greeting page briefly loads before redirecting to custom homepage

### ISSUE THREAD

**Issue by albertoperdomo** [metabase/metabase#35454]
### Describe the bug

Setting a custom homepage to a dashboard still renders the default homepage with the greeting briefly before redirecting to the homepage.

### To Reproduce

1. Set custom homepage
2. Open / in the browser. You will briefly see the default homepage with the greeting before being redirected to the dashboard

Loom with repro:  https://www.loom.com/share/c6f9c10efb9d46ffb3cb054fa0b11345

### Expected behavior

You are taken directly to the dashboard before first rendering the default homepage

### Information about your Metabase installation

```JSON
- Metabase v1.47.6, Built on 2023-10-26
```

### Severity

Annoying UX, but no impact otherwise

**Comment by madmaxmatze** [comment]
I would like to add here:
 - I think the homepage should not be a redirect at all, but stay on the home URL
 - I want the collection tree on the left side still be closed or whatever the default is when looking at the homepage. So not jump to the specific dashboard deep in a big collection tree.
 - I don't want the headline of the dashboard on the homepage, or any of the tool icons on the right
 - Basically, I only want to add some custom cards to the basic homepage, without any other change
